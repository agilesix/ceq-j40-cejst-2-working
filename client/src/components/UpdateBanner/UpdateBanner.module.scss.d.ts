declare namespace UpdateBannerNamespace {
  export interface IUpdateBannerScss {
    updateBannerAlert: string;
    updateBannerContainer: string;
    updateBanner: string;
    updatePillIcon: string;
    updateHeading: string;
  }
}

declare const UpdateBannerScssModule: UpdateBannerNamespace.IUpdateBannerScss & {
  /** WARNING: Only available when `css-loader` is used without `style-loader` or `mini-css-extract-plugin` */
  locals: UpdateBannerNamespace.IUpdateBannerScss;
};

export = UpdateBannerScssModule;
