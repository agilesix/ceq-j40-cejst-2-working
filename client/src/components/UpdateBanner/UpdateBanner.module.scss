@use '../../styles/design-system.scss' as *;

.updateBannerContainer {
  @include u-bg("gray-cool-2");
  @include u-padding-bottom(1);
  @include u-padding-top(1);
}

.updateBanner {
  max-width: 70.25rem; // Needs this exact value to align with GovBanner
  @include u-font("body", "3xs");
  @include u-line-height('ui', 2);
  @include u-margin-left(auto);
  @include u-margin-right(auto);

  @include at-media-max("desktop-lg") {
    @include u-margin-left(4);
    @include u-margin-right(4);
  }

  @include at-media-max("desktop") {
    @include u-margin-left(2);
    @include u-margin-right(2);
  }
}

.updateBannerAlert {
  @include u-margin-top(2);
}
