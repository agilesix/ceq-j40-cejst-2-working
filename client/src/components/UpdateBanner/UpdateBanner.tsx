import React from 'react';

import {Alert} from '@trussworks/react-uswds';
import * as COMMON_COPY from '../../data/copy/common';
import * as styles from './UpdateBanner.module.scss';

export const cutoffDate = new Date(2025, 0, 19); // 2025-01-19

const UpdateBanner = () => {
  const currentDate = new Date();

  // show large version before cutoff date
  const showLarge = currentDate < cutoffDate;

  if (showLarge) {
    return (
      <div className={styles.updateBanner}>
        <Alert type={'info'} headingLevel={'h1'} heading={COMMON_COPY.UPDATE_BANNER_HEADING_LARGE}
          className={styles.updateBannerAlert}>
          {COMMON_COPY.UPDATE_BANNER_CONTENT_LARGE}
        </Alert>
      </div>
    );
  } else {
    return (
      <div className={styles.updateBannerContainer}>
        <div className={styles.updateBanner}>
          {COMMON_COPY.UPDATE_BANNER_CONTENT_SMALL}
        </div>
      </div>
    );
  }
};

export default UpdateBanner;
