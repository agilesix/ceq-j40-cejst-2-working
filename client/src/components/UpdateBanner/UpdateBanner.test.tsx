import {render} from '@testing-library/react';
import * as React from 'react';
import {LocalizedComponent} from '../../test/testHelpers';
import UpdateBanner, {cutoffDate} from './UpdateBanner';

describe('rendering of the UpdateBanner', () => {
  const renderBanner = (mockCurrentDate: Date) => {
    // make this testable at different dates so we can check large/small versions
    jest.useFakeTimers()
        .setSystemTime(mockCurrentDate);

    const result = render(
        <LocalizedComponent>
          <UpdateBanner />
        </LocalizedComponent>,
    ).asFragment();

    jest.useRealTimers();
    return result;
  };

  it('renders large version before cutoff date', () => {
    const beforeCutoff = new Date(cutoffDate);
    beforeCutoff.setDate(beforeCutoff.getDate() - 7);
    expect(renderBanner(beforeCutoff)).toMatchSnapshot();
  });

  it('renders small version at cutoff date', () => {
    expect(renderBanner(cutoffDate)).toMatchSnapshot();
  });
});
