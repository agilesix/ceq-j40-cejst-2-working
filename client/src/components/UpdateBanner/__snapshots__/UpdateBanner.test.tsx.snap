// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering of the UpdateBanner renders large version before cutoff date 1`] = `
<DocumentFragment>
  <div>
    <div
      class="usa-alert usa-alert--info"
      data-testid="alert"
    >
      <div
        class="usa-alert__body"
      >
        <h1
          class="usa-alert__heading"
        >
          Version 2.0 of the tool is now available
        </h1>
        <p
          class="usa-alert__text"
        >
          The Council on Environmental Quality (CEQ) made the 2.0 version of the tool available on Dec 19, 2024.
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`rendering of the UpdateBanner renders small version at cutoff date 1`] = `
<DocumentFragment>
  <div>
    <div>
      <strong>
        This tool has been updated.
      </strong>
       The 2.0 version of the tool was released on Dec 19, 2024.
    </div>
  </div>
</DocumentFragment>
`;
